### ISSUE TITLE
Server still crashes when generating loot

### ISSUE THREAD

**Issue by ArmonicPvP** [GC-spigot/AdvancedEnchantments#4271]
### Describe the bug

Server crashes when generating loot

https://hastebin.com/share/uyedivikur.markdown

https://github.com/GC-spigot/AdvancedEnchantments/issues/4233

### How to reproduce

Generate loot in 1.20.4

### "/ae plinfo" link

https://paste.md-5.net/jawahixibe

**Comment by SirJben** [comment]
Helloo @ArmonicPvP did you modified your loot file? If so, send it here.

**Comment by ArmonicPvP** [comment]
`# Configuration file for random loot generated in world and villager trades
#  Random locations include:
#   * Any chests generated in wild, e.g. mob spawner rooms, mineshaft etc.
#  Configure Villager Trades. Villagers can trade custom enchantment books and admin items, e.g. white scrolls, orbs, dusts.

# ------ Random Loot Section ------ #
randomLoot:
  # Should random loot be generated in world?
  randomLootEnabled: true

  # Maximum number of custom items to be generated in a random loot container
  maxItemsPerContainer: 2

  # Chance of any item(s) being generated
  chanceOfItems: 33

  # Blacklisted groups, these group books cannot generate in random loot
  blacklisted-groups:
    - 'STAFF'

  # Chances format:
  # ITEM_NAME: Chance (0 to 100)
  chances:
    SLOT_INCREASER: 0
    WHITE_SCROLL: 0
    MYSTERY_DUST: 0
    SECRET_DUST: 0
    MAGIC_DUST: 0
    BLACK_SCROLL: 0
    RANDOMIZER: 0
    RENAME_TAG: 0
    BLOCKTRAK: 0
    FISHTRAK: 0
    STATTRAK: 0
    MOBTRAK: 0
    SOUL_TRACKER: 5
    SOUL_GEM: 0
    TRANSMOG_SCROLL: 0
    HOLY_WHITE_SCROLL: 0
    ARMOR_ORB: 0
    WEAPON_ORB: 0
    TOOL_ORB: 0
    RANDOM_ENCHANT_BOOK: 95

# ------ Villager Trades Section ------ #
villagerTrades:
  # Should villagers be able to trade custom books / items?
  villagerTradesEnabled: false

  # Should villagers trade admin items as well? E.g. Whitescroll, Blackscroll, Magic Dust
  # If set to false, they will only trade custom enchantment books
  tradeAllLoot: false

  # Chance of any item(s) being added as a trade
  chanceOfItems: 0

  # Blacklisted groups, these group books cannot be in villager trades
  blacklisted-groups:
    - 'SOUL'

  # Chances format:
  # ITEM_NAME: Chance (0 to 100)
  chances:
    SLOT_INCREASER: 5
    WHITE_SCROLL: 12
    MYSTERY_DUST: 0
    SECRET_DUST: 15
    MAGIC_DUST: 12
    BLACK_SCROLL: 7
    RANDOMIZER: 10
    RENAME_TAG: 8
    BLOCKTRAK: 3
    FISHTRAK: 3
    STATTRAK: 3
    MOBTRAK: 3
    SOUL_TRACKER: 2
    SOUL_GEM: 15
    TRANSMOG_SCROLL: 8
    HOLY_WHITE_SCROLL: 4
    ARMOR_ORB: 2
    WEAPON_ORB: 2
    TOOL_ORB: 2
    RANDOM_ENCHANT_BOOK: 100
`

**Comment by GC-spigot** [comment]
Does the default configuration crash for you or is this modified?

**Comment by ArmonicPvP** [comment]
Default configuration of the lootConfiguration.yml or the whole plugin? I've modified most of the configs.

**Comment by ThomasWega** [comment]
try with default loot config first. If it still crashes try with everything fresh

**Comment by ArmonicPvP** [comment]
lootConfiguration.yml does not spawn randomized loot by default. I will enable it and test.

**Comment by ArmonicPvP** [comment]
Still happens, even with completely default settings. I've attached my config settings as well. I am going to see if I can figure out what structure (or generation) is causing the issue.

[config.zip](https://github.com/GC-spigot/AdvancedEnchantments/files/15313883/config.zip)

**Comment by ArmonicPvP** [comment]
Does debug show when loot is generated in chests?

**Comment by ArmonicPvP** [comment]
Got a massive one here: https://hastebin.com/share/pofiqeqapu.markdown

I went and counted the amount of chests in that spot. There are none. I checked the seed map. There are no structures around the area I generated. I expanded the area to instead of my set render distance to the server set render distance since I am unsure if paper only sends the chunks the player needs but generates the server distance chunks or if that is not the case. When doing so, I find these structures below: 
![image](https://github.com/GC-spigot/AdvancedEnchantments/assets/24983808/ae7ed41a-a14b-4f4e-ab0b-cb85658dea0c)

For testing, I set the maxItemsPerContainer to 18, 100% chance, and 95% random_enchant_book and 5% soul_tracker.

Here is the seed: 3571388013154003028

As a note, I am using Terralith, Incendium, and Structory. AdvancedEnchantments generates custom loot inside of the custom loot chests in all three of these datapacks.

Here is what I noticed: 

- One ocean ruin was supposed to generate, but it did not. Therefore, there was no chest at its location. A terralith generation was at this location in the sky. /tp 19880 ~ -16152
- One shipwreck was supposed to generate, but it did not. Therefore, there was no chest at its location. A terralith generation was at this location in the sky. /tp 19864 ~ -16232
- One mineshaft was supposed to generate, but did not. Therefore, there were no chests at its location. There was no terralith generation at this location. /tp 19880 ~ -16392

My assumption is that an error appears when loot cannot be generated because there is no chest there when it expects there to be one and attempts to find it and lags the server.
